modules/ROOT/pages/architecture/hexagonale_architecture.adoc

```diff
@@ -0,0 +1,218 @@
+Each interface should clearly mark the use case that contains only one method.
+Use cases from the interface might be grouped logically in the use case implementation class.
+====
+
+| core.port.out
+| Outbound ports decouple the interaction with external systems. 
+This might include other services that are called, files that are written, databases, event streaming and everything the application is actively triggering outside of the core.
+Outbound ports should describe the business need for the communication (e.g. StoreReservationPort). How this is then realized depends on the adapter that implements it. 
+This way a technology can be easily replaced. 
+For example storing the reservation could be be realized in a first prototype by writing the objects to a file. 
+Later it could be replaced with a database.
+The core logic would be untouched by that.
+
+| [optional] core.service
+| Services can be considered as business helper classes. 
+They provide a reusable part of the applications business logic that is used by multiple use cases or that helps to structure the application in a logical way.
+Services are optional as they can be used, when there's a real need. Usually a use case should contain the business logic.
+
+| adapter
+a| Adapters connect the application core to the surrounding context. They have the following tasks:
+
+* Implement a specific protocol to connect to the context. E.g REST, JDBC, MQTT, ...
+* Maintain a data model that is necessary to communicate with the context
+* Translate the domain model from the core to that model or vice versa
+* Handle protocol specific errors
+* Log the interaction with the surrounding context
+
+| adapter.in
+| Inbound adapters specify connection points for everything that can trigger the business logic. 
+That might be interfaces (HTML, RPC, etc), Message Consumers or schedulers for batch processing.
+Inside the adapters further packages are differentiating the protocols that is used (e.g. `.rest`). 
+
+| adapter.out
+| Outbound adapters define outgoing connections where the application actively interacts with context outside. 
+That can be database connections, file operations, API calls, message producing and many more.
+Inside the adapters further packages are differentiating the protocols that is used (e.g. `.jpa`).
+|===
+
+
+== Anemic vs Rich domain models
+====
+"In a rich domain model, as much of the domain logic as possible is implemented within the entities at the core of the application.
+The entities provide methods to change state and only allow changes that are valid according to the business rules. [...]
+In an “anemic” domain model, the entities themselves are very thin.
+They usually only provide fields to hold." <<Hombergs21>>
+====
+
+Considering java as an object oriented language it feels natural to implement business logic inside the entities themselves. 
+In large scale application with mixed skilled people it's proposed to not use rich domain models.
```

Comment:
It's obvious that there are `mixed skilled people`, I would leave that out here. It sounds a bit offensive.